userspace/libsinsp/parsers.cpp

```diff
@@ -2655,14 +2717,52 @@ void sinsp_parser::parse_connect_enter(sinsp_evt *evt){
         return;
     }
 
-    packed_data = (uint8_t*)parinfo->m_val;
+	packed_data = (uint8_t*)parinfo->m_val;
 
-    fill_client_socket_info(evt, packed_data);
+	uint8_t family = *packed_data;
+
+	if(family == PPM_AF_INET)
+	{
+		evt->m_fdinfo->m_type = SCAP_FD_IPV4_SOCK;
+		memcpy(&evt->m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip, packed_data + 1, sizeof(uint32_t));
+		memcpy(&evt->m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dport, packed_data + 5, sizeof(uint16_t));
```

Comment:
What about something easier to readers, like:
```
packed_data = (uint8_t*)parinfo->m_val;
uint8_t family = *packed_data;
packed_data++;
if(family == PPM_AF_INET)
{
	evt->m_fdinfo->m_type = SCAP_FD_IPV4_SOCK;
	memcpy(&evt->m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip, packed_data, sizeof(uint32_t));
        packed_data += sizeof(uint32_t);
	memcpy(&evt->m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dport, packed_data, sizeof(uint16_t));
}
else if {...}
else {...}

packed_data = (uint8_t*)parinfo->m_val;
if(m_track_connection_status && m_fd_listener)
{
    m_fd_listener->on_connect(evt, packed_data);
}
...
```

I mean, i can follow the code better this way; perhaps it's just me :D

Reply:
I see your point @FedeDP , as this would get rid of all those magic numbers. However, this entire file uses this way of doing things and having a single function that gives a different meaning to `packed_data` or generally does things differently might be confusing for the reader. I agree that some refactor here is needed (also to get rid of potential UB in other functions...), and would be great for another PR, if you agree.

Reply:
+1 from me! Thanks!